docs/root/api-v1/route_config/route.rst

```diff
@@ -192,6 +193,11 @@ priority
   happen if all the headers in the route are present in the request with the same values (or based
   on presence if the ``value`` field is not in the config).
 
+:ref:`headers <config_http_conn_man_route_table_route_query_args>`
+  *(optional, array)* Specifies a set of query args in the :path on which the route should match.
+  The router will check the request's headers against all the specified query args in the route
+  config.  A match will happen if all the query args match.
+
```

Comment:
You should not need to add this to v1, unless you think you really need this in V1 as opposed to v2 only.